### ISSUE TITLE
Enable HTTPS in dockerized Wallabag

### ISSUE THREAD

**Issue by j0k3r** [wallabag/docker#77]
_From @Suncatcher on July 11, 2017 15:49_

### Issue details
How can I enable HTTPS on Docker installation on Wallabag? Didn't find any decent documentation on that. I see that Docker container uses nginx for Wallabag. Should I enable HTTPS in nginx config inside container?

### Environment

* wallabag version that exhibits the issue: 2.2.2
* How did you install wallabag: Docker
* Last wallabag version that did not exhibit the issue: None
* php version: 7.0.16
* OS: AWS AMI Linux 2017.03
* type of hosting (shared or dedicated): AWS EC2 T2 Micro
* which storage system you choose at install (SQLite, MySQL/MariaDB or PostgreSQL): Postgre 9.6.2

### Steps to reproduce/test case

Trying to enable HTTPS on Docker Wallabag

_Copied from original issue: wallabag/wallabag#3275_

**Comment by j0k3r** [comment]
_From @Suncatcher on July 31, 2017 12:51_

Wow!!! Almost a month and no answers. Nobody can say how to set up a HTTPS?

**Comment by j0k3r** [comment]
_From @Suncatcher on July 31, 2017 12:52_

Is it Docker-specific setting or the way is the same for all versions?

**Comment by j0k3r** [comment]
I think Google will be your best friend for that case.
Few links:
- https://github.com/docker-library/wordpress/issues/46#issuecomment-311275059
- https://tech.acseo.co/sites-https-docker-nginx-lets-encrypt/
- https://github.com/MarvAmBass/docker-nginx-ssl-secure

**Comment by j0k3r** [comment]
_From @Suncatcher on July 31, 2017 13:15_

The first link is about Apache, and the latter is about securing separate nginx Docker container. In our case nginx is included in wallabag image. Is it?

**Comment by j0k3r** [comment]
_From @Suncatcher on July 31, 2017 15:20_

I see [here on Dockerhub](https://hub.docker.com/r/wallabag/wallabag/) we have SSL settings sample for **nginx**:

    server {
            listen 443;
            server_name wallabag.foo.bar;
    
        ssl on;
            ssl_certificate /etc/letsencrypt/live/wallabag.foo.bar/fullchain.pem;
            ssl_certificate_key /etc/letsencrypt/live/wallabag.foo.bar/privkey.pem;
    
        location / {
            proxy_pass http://wallabag;
            proxy_set_header X-Forwarded-Host $server_name;
                    proxy_set_header X-Forwarded-Proto https;
                    proxy_set_header X-Forwarded-For $remote_addr;
        }
    
    }

However, where should we put it? In **Docker-file**? In **docker-compose**? Somewhere inside wallabag container? In running state or in stopped?
HTTPS thing should be somehow highlighted in documentation.

**Comment by j0k3r** [comment]
This should goes to the docker-composer of wallabag/docker.
In fact, I'll move that issue in the correct repo.

**Comment by Suncatcher** [comment]
Do you have many containers? What is the advantage of traefic?

**Comment by nsteinmetz** [comment]
Yes, a dozen at least. 

Traefik is interesting as it integrates docker containers (but also other such as kubernetes, consul, etc) on the fly and provides ssl endpoint (letsencrypt or traditional certificates) and can even be used as traditional reverse proxy.

**Comment by xsteadfastx** [comment]
@j0k3r this should go into a nginx configuration. this is only for the reverse proxy. making wallabag public is a thing of a reverse proxy using a plain http connection to the container in my opinion. but lets discuss it :)

**Comment by Suncatcher** [comment]
in a Wallabag container nginx acts like a reverse proxy by default?

**Comment by xsteadfastx** [comment]
@Suncatcher nope... you should use a reverse proxy for accessing the wallabag container... if you really want to make it public available. there you can you traefik or nginx or something for ssl and stuff. the wallabag image is just for accepting the connection.

**Comment by kedare** [comment]
I highly recommend you Traefik for this as reverse proxy: https://docs.traefik.io/user-guide/docker-and-lets-encrypt/

**Comment by rinokeros** [comment]
> I highly recommend you Traefik for this as reverse proxy: https://docs.traefik.io/user-guide/docker-and-lets-encrypt/

I followed the instruction in your link. But unfortunately I was not able to make it work.
I am a newbie with docker as well... thanks for your help.
There is no real error message. In the browser I get 'gateway timeout'. 

```
version: '3'

networks:
  web:

volumes:
  traefik-acme:
  wallabag-maria-db:

services:
  wallabag:
    image: wallabag/wallabag
    networks:
      - web
      - default
    expose:
      - "80"
    labels:
          traefik.backend: "wallabag"
          traefik.docker.network: "traefik_default"
          traefik.docker.network: "web"
          traefik.frontend.rule: "Host:my.domain.com"
          traefik.enable: "true"
          traefik.port: "80"
          traefik.default.protocol: "http"
    volumes:
      - /opt/wallabag/images:/var/www/wallabag/web/assets/images
    environment:
      - MYSQL_ROOT_PASSWORD=wallaroot
      - SYMFONY__ENV__DATABASE_DRIVER=pdo_mysql
      - SYMFONY__ENV__DATABASE_HOST=db
      - SYMFONY__ENV__DATABASE_PORT=3306
      - SYMFONY__ENV__DATABASE_NAME=wallabag
      - SYMFONY__ENV__DATABASE_USER=wallabag
      - SYMFONY__ENV__DATABASE_PASSWORD=pass
      - SYMFONY__ENV__MAILER_HOST=127.0.0.1
      - SYMFONY__ENV__MAILER_USER=~
      - SYMFONY__ENV__MAILER_PASSWORD=~
      - SYMFONY__ENV__FROM_EMAIL=wallabag@example.com
      - SYMFONY__ENV__FOSUSER_REGISTRATION=false
      - SYMFONY__ENV__DOMAIN_NAME=my.domain.com
  db:
    image: mariadb
    environment:
      - MYSQL_ROOT_PASSWORD=wallaroot
    volumes:
      - /opt/wallabag/data:/var/lib/mysql
  redis:
    image: redis:alpine
```

**Comment by nsteinmetz** [comment]
You need to add traefik to the related network

See https://docs.traefik.io/configuration/backends/docker/#using-docker-compose and the message at the end of the page:

```
When running inside a container, Træfik will need network access through:

docker network connect <network> <traefik-container>
```

**Comment by rbcolom** [comment]
@nsteinmetz I also use traefik for all of my containers. But with wallabag I encountered an issue, when I try:
`curl -I https://my.domain`
I get:

> HTTP/2  302 
> cache-control: no-cache, private
> content-type: text/html; charset=UTF-8
> date: Wed, 05 Dec 2018 11:51:54 GMT
> location: http://my.domain
> server: nginx
> set-cookie: PHPSESSID=71a90d0ec85a887982f2adb7785f4770; path=/; HttpOnly
> strict-transport-security: max-age=315360000; includeSubdomains; preload
> vary: Accept-Encoding
> x-content-type-options: nosniff
> x-frame-options: DENY
> x-powered-by: PHP/7.2.10
> x-xss-protection: 1; mode=block

This is also giving me headaches with the android app telling me: "wallabag service not found -- check url".
Any idea on why this is happening?

**Comment by nsteinmetz** [comment]
@rbcolom 

Seems you have a redirect from https to http ?!

Can you share your `docker-compose.yml` for wallabag and/or traefik and your `traefik.toml` so that I can review and compare it to mine ?

Is your service available on the http url ?

**Comment by rbcolom** [comment]
@nsteinmetz 

Certainly, here is the docker-compose:
```
  wallabag:
    image: wallabag/wallabag
    container_name: wallabag
    healthcheck:
      test: ["CMD", "curl" , "-f", "http://localhost:80"]
      interval: 30s
      timeout: 10s
      retries: 5
    environment:
      - TZ=it-IT
      - PUID=1000
      - PGID=1000
      - "SYMFONY__ENV__FOSUSER_REGISTRATION=false"
      - "SYMFONY__ENV__DOMAIN_NAME=https://my.domain"
      - "SYMFONY__ENV__MAILER_HOST=~"
      - "SYMFONY__ENV__MAILER_USER=~"
      - "SYMFONY__ENV__MAILER_PASSWORD=~"
      - "SYMFONY__ENV__FROM_EMAIL=~"
    
    networks:
      - traefik_proxy
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /home/roberto/DockerConf/wallabag/images:/var/www/wallabag/web/assets/images
      - /home/roberto/DockerConf/wallabag/data:/var/www/wallabag/data
    labels:
      - "traefik.enable=true"
      - "traefik.backend=wallabag"
      - "traefik.frontend.rule=Host:my.domain"
      - "traefik.port=80"
      - "traefik.protocol=http"
      - "traefik.docker.network=traefik_proxy"
      - "traefik.frontend.headers.SSLRedirect=true"
      - "traefik.frontend.headers.STSSeconds=315360000"
      - "traefik.frontend.headers.browserXSSFilter=true"
      - "traefik.frontend.headers.contentTypeNosniff=true"
      - "traefik.frontend.headers.forceSTSHeader=true"
      - "traefik.frontend.headers.SSLHost=my.domain"
      - "traefik.frontend.headers.STSIncludeSubdomains=true"
      - "traefik.frontend.headers.STSPreload=true"
      - "traefik.frontend.headers.frameDeny=true"
      - "traefik.frontend.redirect.entryPoint=https"
```
and here is the relevant traefik.toml part:
```
# Force HTTPS
[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
    entryPoint = "https"
  [entryPoints.https]
  address = ":443"
  compress = true
    [entryPoints.https.tls]
```
For all of my services I use a http to https redirect.
BTW thanks for the help, this is driving me nuts.

**Comment by nsteinmetz** [comment]
@rbcolom take care about your environment variables - you should not have quotes - cf https://docs.docker.com/compose/compose-file/#environment

For my wallabag docker-compose.yml:

```
version: '2'
services:
  wallabag:
    image: wallabag/wallabag
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=XXX
      - POPULATE_DATABASE=False
      - SYMFONY__ENV__DATABASE_DRIVER=pdo_pgsql
      - SYMFONY__ENV__DATABASE_HOST=db
      - SYMFONY__ENV__DATABASE_PORT=5432
      - SYMFONY__ENV__DATABASE_NAME=wallabag
      - SYMFONY__ENV__DATABASE_USER=wallabag
      - SYMFONY__ENV__DATABASE_PASSWORD=XXX
      - SYMFONY__ENV__DOMAIN_NAME=https://wallabag.domain.tld
    labels:
      - "traefik.backend=steinmetz-wallabag"
      - "traefik.frontend.rule=Host:wallabag.domain.tld"
      - "traefik.port=80"
      - "traefik.protocol=http"
      - "traefik.frontend.entryPoints=http,https"
    depends_on:
      - db
      - redis
    restart: always
  db:
    image: postgres:9.6-alpine
    environment:
      - POSTGRES_PASSWORD=XXX
    volumes:
      - /srv/www/steinmetz.fr/w/db:/var/lib/postgresql/data
    labels:
      - "traefik.enable=false"
    restart: always
  redis:
    image: redis:alpine
    labels:
      - "traefik.enable=false"
    restart: always
```

How do you launch traeik ? I don't see the docker integration/activation ?

On my side, I deploy traefik as a binary - not as a docker container - managed by systemd and with follwoing `traefik.toml`:

```
defaultEntryPoints = ["http", "https"]
traefikLogsFile = "/var/log/traefik.log"
logLevel = "ERROR"

[accessLog]
filePath = "/var/log/traefik_access.log"

[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
      entryPoint = "https"
  [entryPoints.https]
  address = ":443"
    [entryPoints.https.tls]

[...]

[docker]
endpoint = "unix:///var/run/docker.sock"
domain = "docker.localhost"
watch = true
```

**Comment by rbcolom** [comment]
Pardon me, I cut out the wallabag section in the docker-compose file.
I tried your configuration (but still using a traefik docker image) on a different machine and I still get the same results.
Can you confirm me that you're able to use the android application?

**Comment by nsteinmetz** [comment]
if you use a traefik docker image, you need to connect traefik to your wallabag network (or the opposite).

I use the android version without any issue.

**Comment by rbcolom** [comment]
Sorry, I didn't explain myself properly. With my current configuration (the one in the comments above) wallabag is up and running, I can access it through my domain without any problem.
What doesn't work is the android app. Investigating a bit, I found out that the POST request wallabag.domain/login ignores the reverse proxy and passing via http.
so when I do:
` curl -I https://my.domain/login`
I get:
`location: http://my.domain/login`

**Comment by rolisz** [comment]
I think the problem is that the Symfony setting for trusting the reverse proxy is not set. Instructions for that are here: https://symfony.com/doc/3.4/deployment/proxies.html

I manually added that in my Docker in the `web/app.php` and it works. That's probably not the best place to add it, but I'm not familiar with Symfony 3 code organization. 

Also, what I've seen in other PHP projects is that they add an environment variable called `trusted_proxies`, which can be used to set this value dynamically (from Docker for example).

**Comment by rbcolom** [comment]
I've upgraded to the 2.1.0 version of the wallabag android app and it is now working. I'm using the latest docker image of wallabag and the reverse proxy is traefik.

**Comment by rolisz** [comment]
But does the redirect from https://my.domain to http://my.domain/login still happen?

**Comment by rbcolom** [comment]
@rolisz Yes, it still occurs, the response is the same. If you need any other info/data, I'll be glad to provided to you.

**Comment by norweeg** [comment]
@rolisz this downgrading when redirecting is causing issues for anyone using the wallabag android app in android 9 FYI  see https://github.com/wallabag/android-app/issues/756, https://github.com/wallabag/wallabag/issues/2273, and https://github.com/wallabag/docker/issues/110

**Comment by norweeg** [comment]
Adding the following traefik frontend redirects via labels on my wallabag container worked for me.  even though wallabag tries to downgrade to http on redirect, my frontend now upgrades it back to https.

```
      - "traefik.frontend.redirect.regex=^http://wallabag.example.com/(.*)"
      - "traefik.frontend.redirect.replacement=https://wallabag.example.com/$$1"
      - "traefik.frontend.redirect.permanent=true"

```

```
norweeg@my-pc$ curl -I https://wallabag.example.com 
HTTP/2 302  
cache-control: no-cache, private
content-type: text/html; charset=UTF-8
date: Sun, 03 Feb 2019 19:23:50 GMT
location: http://wallabag.example.com/login
referrer-policy: no-referrer 
server: nginx
set-cookie: PHPSESSID=574fb5e6fb8c7a092532a93c98294d2b; path=/; HttpOnly
strict-transport-security: max-age=15552000 
x-powered-by: PHP/7.2.13

norweeg@mypc$ curl -I http://wallabag.example.com/login                                                                                                                                                                  
HTTP/1.1 307 Temporary Redirect 
Location: https://wallabag.example.com:443/login
Date: Sun, 03 Feb 2019 19:24:01 GMT
Content-Length: 18
Content-Type: text/plain; charset=utf-8

norweeg@my-pc$ curl -I https://wallabag.example.com:443/login                                                                                                                                                   
HTTP/2 200  
cache-control: no-cache, private
content-type: text/html; charset=UTF-8 
date: Sun, 03 Feb 2019 19:24:22 GMT
referrer-policy: no-referrer
server: nginx 
set-cookie: PHPSESSID=e5b601c4811270133e530c0d986e7280; path=/; HttpOnly
strict-transport-security: max-age=15552000
x-powered-by: PHP/7.2.13 
```

**Comment by Chostakovitch** [comment]
@norweeg I still have the problem even with your suggestion. 
I found that `traefik.frontend.headers.SSLRedirect` have the same behaviour (308 + correct HTTPS URL), but in both cases the Android app keeps throwing the error message.

**Comment by norweeg** [comment]
@Chostakovitch  huh.. that's strange because I don't touch that option, but I stopped getting the error once my wallabag frontend sent a http 301 redirect instead of 302 redirect from http to https (the redirect at the entrypoint level wasn't sufficient).  I probably forgot something in my earlier post, so here's my docker-compose service template for my wallabag service that has worked for me:

```yaml
version: '3.5'
services:
  wallabag:
    image: wallabag/wallabag:latest
    container_name: wallabag
    networks:
      - wallabag
    restart: unless-stopped
    env_file:
      - ../configs/wallabag.env
    expose:
      - "80"
    labels:
      - traefik.enable=true
      - traefik.backend=wallabag
      - traefik.frontend.rule=Host:${WALLABAG_DOMAIN}
      - "traefik.frontend.redirect.regex=^http://${WALLABAG_DOMAIN}/(.*)"
      - "traefik.frontend.redirect.replacement=https://${WALLABAG_DOMAIN}/$$1"
      - traefik.frontend.redirect.permanent=true
      - traefik.frontend.passHostHeader=true
      - traefik.frontend.whiteList.useXForwardedFor=true
      - traefik.frontend.headers.SSLProxyHeaders=X-Forwarded-Proto:https
      - traefik.docker.network=wallabag
      - traefik.port=80
      - com.centurylinklabs.watchtower.enable=true
    depends_on:
      - wallabag-db
      - wallabag-redis
    healthcheck:
      test: ["CMD-SHELL", "curl -f http://wallabag:80/login > /dev/null || exit 1"]
      interval: 15s
      timeout: 10s
      retries: 5
      start_period: 15s
    volumes:
      - wallabag-images:/var/www/wallabag/web/assets/images
      - wallabag-data:/var/www/wallabag/data
      - wallabag-logs:/var/www/wallabag/var/logs
```

**Comment by Chostakovitch** [comment]
@norweeg Thanks for your help. I'm still getting the same behaviour, even with your exact parameters.
When I try to `curl` my instance, I have the same results as you.
However, when I connect from Android app, I can see (in `docker-compose logs`) this line : 

`wallabag       | 172.20.0.6 - - [27/Feb/2019:19:46:13 +0000] "GET / HTTP/1.1" 302 206 "-" "okhttp/3.12.1"`

And then the error message on the app. Can you confirm than you are also getting a 302 inside of the container ? What is the traefik version that you use ?

Thanks :)

**Comment by norweeg** [comment]
@Chostakovitch using Traefik V1.7.9 / MAROILLES.  when you say get the 302 "inside" the container, which container are you referring to?

**Comment by diegombeltran** [comment]
I'm using the same config as you wrote and still get the CLEARTEXT error. When curl to https://wallabag.mydomain.com it still redirects to http://

**Comment by diegombeltran** [comment]
Mmmmh, using Traefik 1.7.9-alpine, Wallabag latest image and clearing data/cache.

There must be something else...

**Comment by norweeg** [comment]
@diegombeltran  oh if you want to know which traefik image:tag I use, I use traefik:latest and have a watchtower instance that automatically updates it, so I always have what's current.  Since it has no external dependencies, I consider it safe to track the latest tag.  Question for you and @Chostakovitch: you have [Traefik configured to handle SSL termination](https://docs.traefik.io/user-guide/docker-and-lets-encrypt/), right?  Also how do you have your Traefik entrypoints configured?  Mine are pretty standard and look like this:

```toml
defaultEntryPoints = ["https","http"]

[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
    entryPoint = "https"
  [entryPoints.https]
  address = ":443"
  [entryPoints.https.tls]

[retry]
```

**Comment by 0xelion** [comment]
I'm having the same issue as @diegombeltran 
I'm running traefik:1.7.9-alpine and wallabag:2.3.6 and even with @norweeg's config i get the CLEARTEXT issue.

**Comment by norweeg** [comment]
@0xelion do you know if the latest tag and the alpine tag are the same image?  what happens if you, for the sake of testing it, use traefik:latest instead.

**Comment by 0xelion** [comment]
@norweeg It isn't, the two images are different and i still have the issue with the latest image.

**Comment by norweeg** [comment]
@0xelion so strange!  I have no explanation for how we can be using the same images and configuration yet get different results

**Comment by elecprog** [comment]
Same issue here. What versions of Android are you on? Because it seems to me as if it's down to Android 9, which by default disallows cleartext connections. (See <https://developer.android.com/guide/topics/manifest/application-element#usesCleartextTraffic>).

**Comment by 0xelion** [comment]
@elecprog I'm on Android 9 (API level 28)

**Comment by elecprog** [comment]
There already appears to be a pull request including this change <https://github.com/wallabag/android-app/pull/762> So let's wait and see.

**Comment by norweeg** [comment]
@elecprog I don't know how I feel about a PR that just allows cleartext transmission of a password.  It disallows it for a reason.  The really ought to fix the Wallabag web application and make it honor the X-Forwarded-For-Proto when redirecting

**Comment by elecprog** [comment]
As far as I can tell it blocks any non-HTTPS connection. On the other hand, we have Let's Encrypt now so that might not be an issue anymore.

**Comment by norweeg** [comment]
@elecprog somone over on the PR made a good point about a bigger, more permanent fix is out of scope and that the PR just makes the behavior on Android >= 9 consistent with Android < 9, still, the wallabag server really ought to know to redirect to https addresses if X-Forwarded-For-Proto is set to 'https' and we, the users, shouldn't have to use kludgy redirects of redirects to work-around the issue

**Comment by dsaoijhlgfdngfsd** [comment]
@norweeg redirection to HTTP instead of HTTPS is not something 'hardcoded' in Wallabag. This behavior originates in Symfony.

I believe solution posted by @rolisz is correct - you need to add your reverse proxy to the trusted proxies. The issue with Wallabag is that you cannot configure trusted proxies in docker compose or anywhere else besides touching the source code itself. 

A "simple" fix would be to change `web/app.php` and add something like this:
```
if ($trustedProxies = $_SERVER['TRUSTED_PROXIES'] ?? false) {
    Request::setTrustedProxies(explode(',', $trustedProxies));
}
```
where `'TRUSTED_PROXIES'` is an environment variable you can define in docker-compose.yml and should be the IP address of your reverse proxy (or the IP-range in which your reverse proxy resides).

Such approach can be found [here](https://github.com/symfony/recipes/pull/182).

Therefore maybe this issue can be rather solved in wallabag/wallabag repo? @j0k3r 

Btw, this redirect from HTTPS -> HTTP (login) -> HTTPS (login) also happens on `https://wallabag.it` :)

**Comment by j0k3r** [comment]
Yeah might be a good idea.
Can you open an issue? Or do the PR? :)

**Comment by norweeg** [comment]
@dsaoijhlgfdngfsd yeah, I did figure that out, but since I am using docker as a container, that limits my ability to edit things like that without building/maintaining my own container.  Is there a way to make Symfony honor the [X-Forwarded-Proto header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Proto) that a proxy or load balancer should add to requests it forwards?  I think that would be preferable than having to control it via an environment variable, especially since the Mozilla calls it the "de-facto standard header for identifying the protocol (HTTP or HTTPS) that a client used to connect to your proxy or load balancer"

**Comment by dsaoijhlgfdngfsd** [comment]
@norweeg you're kinda mixing up two things - Symfony **does** honor the _X-Forwarded-*_ headers provided they are sent from a **trusted** reverse proxy. Which proxy is trusted? This is usually unknown at the time of building a docker container, thus the IP of the trusted proxies cannot be hardcoded in the docker container. Hence the use of **environment variables** which can tell Symfony which proxies to trust (as a side note - environment variables are used commonly in docker containers, so each container instance can be "tailored" to the environment in which it is deployed). In short - it's not either honoring HTTP headers or using environment variable - in this case you need to use both.

To make this comment actually useful, let me show you a way how to tell Symfony which reverse proxy to trust using the existing Wallabag docker container, although this is dirty and I **do not** recommend it in the long term.

Let's work with this docker-compose record for wallabag:
```
wallabag:
    container_name: wallabag
    image: wallabag/wallabag
    environment:
      - ...
    volumes:
      - ...
```
Your record will be similar, with environment and volumes tailored to your environment. You can change it to:
```
wallabag:
    container_name: wallabag
    image: wallabag/wallabag
    environment:
      - ...
      - TRUSTED_PROXIES=10.1.2.3  # this is IP of your reverse proxy
    volumes:
      - ...
      - $SOME_PATH_ON_YOUR_SERVER/app.php:/var/www/wallabag/web/app.php
```
You need to create app.php on your machine and its content should be:
```
<?php

use Symfony\Component\HttpFoundation\Request;

/**
 * @var Composer\Autoload\ClassLoader
 */
$loader = require __DIR__.'/../app/autoload.php';
include_once __DIR__.'/../var/bootstrap.php.cache';

$kernel = new AppKernel('prod', false);
$kernel->loadClassCache();
//$kernel = new AppCache($kernel);

// When using the HttpCache, you need to call the method in your front controller instead of relying on the configuration parameter
//Request::enableHttpMethodParameterOverride();
if ($trustedProxies = $_SERVER['TRUSTED_PROXIES'] ?? false) {
    Request::setTrustedProxies(explode(',', $trustedProxies));
}
$request = Request::createFromGlobals();
$response = $kernel->handle($request);
$response->send();
$kernel->terminate($request, $response);
```
Should work, haven't tried it though.

**Comment by dsaoijhlgfdngfsd** [comment]
@j0k3r will try to do the PR - not a php programmer though so it might be painful to review :)

**Comment by norweeg** [comment]
@dsaoijhlgfdngfsd Ah, I see.  One problem, though....  Docker assigns the ip to my proxy, so I can't know what it will be when I docker-compose up nor can I access its ip via an environment variable.  It's entirely up to what the docker daemon decides the network range for my wallabag network will be and what ip my proxy has on that network.  Is it possible to resolve the address of your trusted proxy by hostname so something like this would work:

```yaml
services:
  proxy:
    image: traefik:latest
    networks:
      - internal
      - external
  wallabag:
    container_name: wallabag
    image: wallabag/wallabag
    environment:
      - ...
      - TRUSTED_PROXIES=proxy #resolved by docker using service or container name
    networks:
      - internal
```

**Comment by dsaoijhlgfdngfsd** [comment]
@norweeg You can force Docker to assign IP addresses from a specific range, as well as force a certain container to have a specific IP. Also, you can also use range of IP addresses in TRUSTED_PROXIES: `TRUSTED_PROXIES=10.0.0.0/8,192.168.0.0/16`.

**Comment by norweeg** [comment]
good to know!  thanks!  somehow I missed that in the docker-compose docs.  Probably the range solution is better.  I'd hate to statically set a container ip only to have it conflict with a service already running

**Comment by blopware** [comment]
Just my 2c.

I'm using nginx, not traefik for reverse proxying.

I had this issue last night and I spent a bit of time figuring out a solution. So I ended up doing a workaround and not a proper solution, but I hope it is still useful.

- Fetch the `/etc/nginx/nginx.conf` file from the docker container
- Add the line `fastcgi_param  HTTPS 'on';` on the php section
- Bind the modified nginx.conf file into the `/etc/nginx/nginx.conf` location
